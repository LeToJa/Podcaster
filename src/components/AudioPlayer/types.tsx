export interface AudioPlayerTypes {
    url: string
    type: string
}