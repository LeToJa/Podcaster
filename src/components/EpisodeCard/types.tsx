export interface EpisodeCardTypes {
    podcastId: string
    episodeId: number
    title: string
    date: string
    duration: string
}