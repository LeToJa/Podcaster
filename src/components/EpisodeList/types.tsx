import { PodcastEpisode } from '../../layout/PodcastLayout/types'

export interface EpisodeListTypes {
    podcastId: string
    episodes: PodcastEpisode[]
}