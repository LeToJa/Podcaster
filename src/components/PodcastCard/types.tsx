export interface PocastCardTypes {
    id: string
    name: string
}