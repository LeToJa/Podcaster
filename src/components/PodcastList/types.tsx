import { PocastCardTypes } from '../PodcastCard/types'

export interface PodcastListTypes {
    podcasts: PocastCardTypes[]
}