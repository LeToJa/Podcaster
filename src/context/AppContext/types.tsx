export interface AppContextTypes {
    loading: boolean
    toggleLoading: (value: boolean) => void
}