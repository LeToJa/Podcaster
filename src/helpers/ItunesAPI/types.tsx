export interface FeedResponseTypes {
    feed: {
        entry: {
            id: {
                attributes: {
                    'im:id': string
                }
            }
            'im:name': { label: string }
            'im:artist': { label: string }
            'im:image': { label: string }[]
        }[]
    }
}

export interface DefaultResponseFetch {
    contents: string
}

export interface PodcastResponseTypes {
    resultCount: number;
    results: (PodcastResultsEntity)[];
}
interface PodcastResultsEntity {
    wrapperType: string;
    kind: string;
    artistId: number;
    collectionId: number;
    trackId: number;
    artistName: string;
    collectionName: string;
    trackName: string;
    collectionCensoredName: string;
    trackCensoredName: string;
    artistViewUrl: string;
    collectionViewUrl: string;
    feedUrl: string;
    trackViewUrl: string;
    artworkUrl30: string;
    artworkUrl60: string;
    artworkUrl100: string;
    collectionPrice: number;
    trackPrice: number;
    collectionHdPrice: number;
    releaseDate: string;
    collectionExplicitness: string;
    trackExplicitness: string;
    trackCount: number;
    trackTimeMillis: number;
    country: string;
    currency: string;
    primaryGenreName: string;
    contentAdvisoryRating: string;
    artworkUrl600: string;
    genreIds?: (string)[] | null;
    genres?: (string)[] | null;
}
