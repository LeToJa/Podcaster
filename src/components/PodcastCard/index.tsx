import { Link } from 'react-router-dom'

import { PocastCardTypes } from './types'

const PodcastCard = ({ id, title, author, artwork }: PocastCardTypes) => <>
    <Link to={`/podcast/${id}`} className='block shadow-xl px-2 py-4'>
        <img className='w-2/5 mx-auto rounded-full' src={artwork} alt={title} />
        <header className='px-6 py-4'>
            <h4 className='font-bold text-xl mb-2 text-center whitespace-nowrap text-ellipsis overflow-hidden'>{title}</h4>
            <h6 className='text-gray-700 text-base text-center whitespace-nowrap text-ellipsis overflow-hidden'>{author}</h6>
        </header>
    </Link>
</>

export default PodcastCard