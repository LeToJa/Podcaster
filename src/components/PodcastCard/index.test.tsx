import { BrowserRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'

import { PocastCardTypes } from './types'

import PodcastCard from '.'

const mockProps: PocastCardTypes = {
    id: '123',
    title: 'Sample Podcast',
    author: 'John Doe',
    artwork: 'https://via.placeholder.com/150'
}

describe('PodcastCard', () => {
    it('renders correctly and matches snapshot', () => {
        const { container } = render(
            <BrowserRouter>
                <PodcastCard {...mockProps} />
            </BrowserRouter>
        )

        expect(container).toMatchSnapshot()
    })

    it('renders the correct title and author', () => {
        const { getByText } = render(
            <BrowserRouter>
                <PodcastCard {...mockProps} />
            </BrowserRouter>
        )

        expect(getByText('Sample Podcast')).toBeTruthy()
        expect(getByText('John Doe')).toBeTruthy()
    })

    it('contains the correct link', () => {
        const { getByRole } = render(
            <BrowserRouter>
                <PodcastCard {...mockProps} />
            </BrowserRouter>
        )

        const link = getByRole('link')
        expect(link).toHaveProperty('href', 'http://localhost:3000/podcast/123')
    })

    it('renders the artwork with the correct src and alt attributes', () => {
        const { getByAltText } = render(
            <BrowserRouter>
                <PodcastCard {...mockProps} />
            </BrowserRouter>
        )

        const image = getByAltText('Sample Podcast')
        expect(image).toHaveProperty('src', 'https://via.placeholder.com/150')
    })
})
