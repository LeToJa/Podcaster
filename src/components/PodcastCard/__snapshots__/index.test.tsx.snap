// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PodcastCard > renders correctly and matches snapshot 1`] = `
<div>
  <a
    class="block bg-gray-50 border-2 border-gray-200 shadow-xl px-2 py-4"
    data-testid="podcast-card"
    href="/podcast/123"
  >
    <img
      alt="Sample Podcast"
      class="w-2/5 mx-auto rounded-full"
      src="https://via.placeholder.com/150"
    />
    <header
      class="px-6 py-4"
    >
      <h4
        class="font-bold text-xl mb-2 text-center whitespace-nowrap text-ellipsis overflow-hidden"
      >
        Sample Podcast
      </h4>
      <h6
        class="text-gray-700 text-base text-center whitespace-nowrap text-ellipsis overflow-hidden"
      >
        John Doe
      </h6>
    </header>
  </a>
</div>
`;
